### PR TITLE
Adds consistency checks for the `HostConfiguration`

### PR DESCRIPTION
**Pull request by bkchr**
Besides that it fixes the chain specs to make the consistency checks happy.

### CODE REVIEW

**Review by rphmeier — APPROVED** [review]

**Comment by rphmeier** [comment]
In the scheduler, we actually interpret `group_rotation_frequency.is_zero()` as 'no rotations'. So I suppose that check can be removed, along with the doc there and in the guide.

**Comment by bkchr** [comment]
> In the scheduler, we actually interpret `group_rotation_frequency.is_zero()` as 'no rotations'. So I suppose that check can be removed, along with the doc there and in the guide.

Fine by me, I thought also about this. Should we also add a check + comment that the `group_rotation_frequency`? Otherwise we probably run into some problems as well?

**Comment by rphmeier** [comment]
> Fine by me, I thought also about this. Should we also add a check + comment that the group_rotation_frequency? 

If you mean adding a comment that a value of `0` for `group_rotation_frequency` is interpreted as 'no rotations', then I think that's a good idea.

File: runtime/parachains/src/configuration.rs [hunk 1]
```diff
@@ -132,6 +142,31 @@ pub struct HostConfiguration<BlockNumber> {
 	pub hrmp_max_message_num_per_candidate: u32,
 }
 
+impl<BlockNumber: Zero> HostConfiguration<BlockNumber> {
+	/// Checks that this instance is consistent with the requirements on each individual member.
+	///
+	/// # Panic
+	///
+	/// This function panicks if any member is not set properly.
```
Comment (shawntabrizi) [inline runtime/parachains/src/configuration.rs:150]:
```suggestion
	/// This function panics if any member is not set properly.
```

**Review by shawntabrizi — APPROVED** [review]

**Comment by bkchr** [comment]
> > Fine by me, I thought also about this. Should we also add a check + comment that the group_rotation_frequency?
> 
> If you mean adding a comment that a value of `0` for `group_rotation_frequency` is interpreted as 'no rotations', then I think that's a good idea.

No, I thought about removing the checks you mentioned and adding a comment that `group_rotation_frequency` > `chain_availability_period`/`thread_availability_period`.

Do we require support for no rotations? However I would propose to do this in another pr.

File: runtime/parachains/src/configuration.rs [hunk 2]
```diff
@@ -297,6 +353,11 @@ decl_module! {
 		#[weight = (1_000, DispatchClass::Operational)]
 		pub fn set_thread_availability_period(origin, new: T::BlockNumber) -> DispatchResult {
 			ensure_root(origin)?;
+
+			if new.is_zero() {
```
Comment (ordian) [inline runtime/parachains/src/configuration.rs:357]: could you update `set_no_show_slots` check to this style for consistency?

File: runtime/parachains/src/configuration.rs [hunk 3]
```diff
@@ -237,6 +323,9 @@ decl_module! {
 		#[weight = (1_000, DispatchClass::Operational)]
 		pub fn set_group_rotation_frequency(origin, new: T::BlockNumber) -> DispatchResult {
 			ensure_root(origin)?;
+
+			ensure!(new >= 1, Error::<T>::InvalidNewValue);
```
Comment (ordian) [inline runtime/parachains/src/configuration.rs:327]: to please the CI here and below
```suggestion
			ensure!(!new.is_zero(), Error::<T>::InvalidNewValue);
```

**Review by ordian — APPROVED** [review]

**Comment by ordian** [comment]
test `group_rotation_info_calculations` is now invalid

**Comment by rphmeier** [comment]
> No, I thought about removing the checks you mentioned and adding a comment that group_rotation_frequency > chain_availability_period/thread_availability_period.

Yeah, I agree with that.

**Pull request merged**